### PR TITLE
Add lightning CI check

### PR DESCRIPTION
**Pull request by albi3ro**
By running the lightning plugin tests during PR's to the PennyLane code, we will be able to catch code that will break the lightning plugin earlier and make sure to have the fix lined up before anything is merged to master.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -21,11 +21,17 @@
 
 <h3>Bug fixes</h3>
 
+<h3>Testing and Continuous Integration/h3>
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:24]:
```suggestion
<h3>Testing and Continuous Integration</h3>
```

**Review by timmysilv — APPROVED** [review]

**Review by mudit2812 — APPROVED** [review]
Thanks Christina

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -405,6 +405,40 @@ jobs:
...
+        run: |
+          python -m pip install --upgrade pip && pip install wheel --upgrade
+          pip install -r requirements-ci.txt --upgrade
+          pip install -r requirements-dev.txt --upgrade
+          python -m pip install --index-url https://test.pypi.org/simple/ pennylane-lightning --pre --no-deps
+
+      - name: Install PennyLane
+        run: |
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
```
Comment (mlxd) [inline .github/workflows/tests.yml:436]: To avoid the issue of the PennyLane wheel installing an overriding the dev/rc versions on Lightning, it would be safest to allow it to install PennyLane and all dependencies first (lightning included) , and then install the newest Lightning on-top of this without additional deps.

tl;dr Swap the order of these two steps.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -405,6 +405,40 @@ jobs:
...
+
+      - name: Run tests
+        run: |
+          python -m pytest pennylane/devices/tests --device=lightning.qubit
```
Comment (mlxd) [inline .github/workflows/tests.yml:440]: Can we add something like `pl-device-test --device lightning.qubit --shots 10000 --skip-ops` or better here?

**Review by mlxd — CHANGES_REQUESTED** [review]
Thanks @albi3ro Just some minor updates here,
